databases/week1/sqlhomeworkweek1.sql

```diff
@@ -0,0 +1,47 @@
+-- 1. Find out how many tasks are in the task table
+SELECT COUNT(id) AS task FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id) FROM task 
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT * FROM task WHERE status_id = 3;
+
+-- 4. Find all the tasks that are not marked as done
+SELECT * FROM task WHERE status_id != 3;
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT * FROM task 
+ORDER BY created DESC;
+
+-- 6. Get the single most recently created task
+SELECT * FROM task 
+ORDER BY created DESC
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date FROM task 
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+--  8. Get the title and status (as text) of all tasks
+SELECT t.title, s.name 
+FROM task t
+JOIN status s
+ON t.status_id = s.id; 
+
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT s.name, COUNT(*) AS number_of_tasks  
+FROM status s 
+LEFT JOIN task t
+ON t.status_id = s.id
+GROUP BY s.id; 
+
+-- 10. Get the names of all statuses, sorted by the status with most tasks first
+SELECT s.name, COUNT(t.s.id) 
+FROM task t
+JOIN status s
+ON t.status_id = s.id
+GROUP BY s.id  -- use this instead of s.name
+ORDER BY COUNT(t.s.id) DESC ; 
```

Comment:
This last query gives me an error now: `Error in query (1054): Unknown column 't.s.id' in 'field list'`

Looks like you might have made a typo by accident. What I meant in the previous comment was actually just to give the count an alias, just like you did in query no 9 `COUNT(*) AS number_of_tasks`

Something like this:
```suggestion
SELECT s.name, COUNT(t.id) AS number_of_tasks
FROM task t
JOIN status s
ON t.status_id = s.id
GROUP BY s.id  -- use this instead of s.name
ORDER BY number_of_tasks DESC ; 
```